### ISSUE TITLE
sql: unexpected numeric output on valid queries

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#7413]
Found via #7409:

```
logic_test.go:863: /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_2.test:6290:                                                                                                                                                                        expected:                 
         444077
 but found:
            442077
logic_test.go:863: /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_2.test:28356:
        expected:
            -33
        but found:
            -49

logic_test.go:863: /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_2.test:48936:
        expected:
            -7168

        but found:
            -9408
```

This should be investigated; the likely cause is that the division is not rounding the result. If that is confirmed, the sqllogictest input must be modified to add the necessary "skipif cockroachdb" directives.

**Comment by knz** [comment]
One more:

```
 /home/kena/go/src/github.com/cockroachdb/sqllogictest/test/random/expr/slt_good_3.test:82676:
```

**Comment by mjibson** [comment]
This is indeed due to the division not rounding. See the simplified case:

```
SELECT 3 / 2 * 2;
```

On Postgres this outputs `2`, on Cockroach `3`.

I think this issue should be closed in favor of #7734.

**Comment by knz** [comment]
Agreed.
